lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -64,6 +66,15 @@ def setup_hosts
       end
     end
 
+    def effectively_ssl?
```

Comment:
Seems like we only need the `effectively_ssl?` predicate for this:

```
      if @api_key && @api_key.value && !effectively_ssl?
        raise(LogStash::ConfigurationError, "Using api_key authentication requires SSL/TLS secured communication using the `ssl => true` option")
      end
```
Which could be moved to the [HttpClientBuilder.setup_api_key](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/882c9dda75819c8411f3b4291a22146179aa20c2/lib/logstash/outputs/elasticsearch/http_client_builder.rb#L154).
We also use it to set `params['ssl']` but that can also be set in the [HttpClientBuilder.setup_ssl](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/882c9dda75819c8411f3b4291a22146179aa20c2/lib/logstash/outputs/elasticsearch/http_client_builder.rb#L109-L110).

Reply:
I'm not sure we should be raising `LogStash::ConfigurationError` from within `HttpClientBuilder` -- that seems like the wrong side of a boundary.